### PR TITLE
vendor: upgrade go-mysql-server

### PR DESCRIPTION
**Pull request by erizocosmico**
Update go-mysql-server with all fixes that have recently been merged.

### CODE REVIEW

**Comment by ajnavarro** [comment]
LGTM after code compiles. OSx is passing but if you check the logs it's throwing an error

**Comment by kuba--** [comment]
Lets rebase and upgrade to the latest version, so it will also fix indexing expressions #238

**Comment by ajnavarro** [comment]
Yep, also don't forget to update documentation pointing to the same go-mysql-server version.

**Comment by erizocosmico** [comment]
Done @ajnavarro 

Perhaps we should open a task to automate the task of updating all that?

**Comment by erizocosmico** [comment]
Yep, I'm gonna upgrade that

**Comment by erizocosmico** [comment]
Done

**Comment by erizocosmico** [comment]
Upgrading to that go-mysql-server broke the tests, fixing.

**Comment by kuba--** [comment]
`table "commits" does not have column "commit_author_when"`

**Comment by erizocosmico** [comment]
Yes, that's why it depends on https://github.com/src-d/go-mysql-server/pull/249 haha

**Comment by erizocosmico** [comment]
Fixed

File: docs/using-gitbase/functions.md [hunk 1]
```diff
@@ -14,4 +14,4 @@ To make some common tasks easier for the user, there are some functions to inter
 
 ## Standard functions
 
-You can check standard functions in [`go-mysql-server` documentation](https://github.com/src-d/go-mysql-server/tree/5620932d8b3ca58edd6bfa4c168073d4c1ff665f#custom-functions).
\ No newline at end of file
+You can check standard functions in [`go-mysql-server` documentation](https://github.com/src-d/go-mysql-server/tree/08b0ce01897f4ed5a2b45ac32bb87cf335d04dd9#custom-functions).
```
Comment (ajnavarro) [inline docs/using-gitbase/functions.md]: that hashes are not the same as in Gopkg.toml

File: docs/using-gitbase/supported-syntax.md [hunk 2]
```diff
@@ -1,3 +1,3 @@
 ## Supported syntax
 
-To see the SQL subset currently supported take a look at [this list](https://github.com/src-d/go-mysql-server/blob/5620932d8b3ca58edd6bfa4c168073d4c1ff665f/SUPPORTED.md) from [src-d/go-mysql-server](https://github.com/src-d/go-mysql-server).
\ No newline at end of file
+To see the SQL subset currently supported take a look at [this list](https://github.com/src-d/go-mysql-server/blob/08b0ce01897f4ed5a2b45ac32bb87cf335d04dd9/SUPPORTED.md) from [src-d/go-mysql-server](https://github.com/src-d/go-mysql-server).
```
Comment (ajnavarro) [inline docs/using-gitbase/supported-syntax.md]: the same.

**Comment by ajnavarro** [comment]
Now we can use the latest commit with https://github.com/src-d/go-mysql-server/pull/250

**Comment by erizocosmico** [comment]
Yup, upgrading to that version as well

**Comment by erizocosmico** [comment]
Done

**Pull request merged**